src/error.c

```diff
@@ -512,6 +512,19 @@ void ErrorMayQuit(const Char * msg, Int arg1, Int arg2)
     Panic("ErrorMayQuit must not return");
 }
 
+/****************************************************************************
+**
+*F  GetValidRNam( <funcname>, <rnam> ) . check if <rnam> is a valid prec rnam
+*/
+UInt GetValidRNam(const char * funcname, Obj rnam)
+{
+    RequirePositiveSmallInt(funcname, rnam, "<rnam>");
+    UInt val = INT_INTOBJ(rnam);
+    RequireArgumentCondition(funcname, rnam, IS_VALID_RNAM(val),
+                             "must be a valid rnam");
```

Comment:
Do we expect this check to be needed in other files? Otherwise, how about just defining it as `static` inside `records.c`? We do similar things for several other specialized argument checker functions.

And then you don't need to export `IS_VALID_RNAM` (and in fact, don't need to define it in the first place).